x-pack/plugins/ml/public/application/jobs/new_job/common/job_creator/job_creator.ts

```diff
@@ -51,6 +51,7 @@ export class JobCreator {
   protected _indexPattern: DataView;
   protected _savedSearch: SavedSearchSavedObject | null;
   protected _indexPatternTitle: IndexPatternTitle = '';
+  protected _indexPatternDisplayName: string = '';
```

Comment:
❓ shall we store the entire `indexPattern` object instead?

Reply:
The index pattern (DataView) is also stored a few lines above this.
I can't recall why the title was originally extracted like this and kept locally. Possibly to avoid accessing the data view every time it was needed.
But I agree that it is unnecessary for `_indexPatternTitle`.
For the `_indexPatternDisplayName` I like how it is only set once in the constructor, otherwise we'd have to run the ternary logic every time it's accessed. Which may be a lot of times because the wizards are not well optimised for react re-renders.